File: javascript/easy/rangeSumQueryImmutable.js
/**
 * @param {number[]} nums
 */
var NumArray = function(nums) {
    this.sums = [];
    let sum = 0;
    for(let num of nums) {
        sum += num;
        this.sums.push(sum);
    }
};

/**
 * @param {number} i
 * @param {number} j
 * @return {number}
 */
NumArray.prototype.sumRange = function(i, j) {
    return (i == 0) ? this.sums[j] : (this.sums[j] - this.sums[i-1]);
};

/**
 * Your NumArray object will be instantiated and called as such:
 * var obj = Object.create(NumArray).createNew(nums)
 * var param_1 = obj.sumRange(i,j)
 */

File: ruby/easy/range_sum_query_immutable.rb
class NumArray

=begin
    :type nums: Integer[]
=end
    def initialize(nums)
        @sums = Array.new
        sum = 0
        nums.each do |num|
            sum += num
            @sums.push(sum)
        end
    end


=begin
    :type i: Integer
    :type j: Integer
    :rtype: Integer
=end
    def sum_range(i, j)
        return @sums[j] - ((i > 0) ? @sums[i-1] : 0)
    end


end

# Your NumArray object will be instantiated and called as such:
# obj = NumArray.new(nums)
# param_1 = obj.sum_range(i, j)
